test/object-store/c_api/c_api.cpp

```diff
@@ -538,17 +538,6 @@ TEST_CASE("C API (non-database)", "[c_api]") {
 #if REALM_ENABLE_SYNC
     SECTION("sync_client_config_t") {
         auto test_sync_client_config = cptr(realm_sync_client_config_new());
-        realm_sync_client_config_set_base_file_path(test_sync_client_config.get(), "some string");
-        CHECK(test_sync_client_config->base_file_path == "some string");
-        realm_sync_client_config_set_metadata_mode(test_sync_client_config.get(),
-                                                   RLM_SYNC_CLIENT_METADATA_MODE_ENCRYPTED);
-        CHECK(test_sync_client_config->metadata_mode ==
-              static_cast<realm::SyncClientConfig::MetadataMode>(RLM_SYNC_CLIENT_METADATA_MODE_ENCRYPTED));
-        auto enc_key = make_test_encryption_key(123);
-        realm_sync_client_config_set_metadata_encryption_key(test_sync_client_config.get(),
-                                                             reinterpret_cast<uint8_t*>(enc_key.data()));
-        CHECK(test_sync_client_config->custom_encryption_key);
-        CHECK(std::equal(enc_key.begin(), enc_key.end(), test_sync_client_config->custom_encryption_key->begin()));
```

Comment:
Would you mind reviving these checks to use against the new "`realm_app_config_set_`..." functions? Thanks!